### PR TITLE
✨ Replace add-helm-code.sh with YQ commands

### PR DESCRIPTION
**Pull request by francostellari**
## Summary

A combination of YQ commands is used to remove the need of using script `add-helm-code.sh` as described in #1756 

## Related issue(s)

Fixes #
#1756

### CODE REVIEW

**Comment by ezrasilvera** [comment]
@francostellari the previous script also had cleanup-tags.  Isn't it being used anymore ?

**Comment by ezrasilvera** [comment]
@francostellari one high level comment. 
1. In general I think we should reduce the usage of Makefile phony targets with complex logic in them and move instead to use utility scripts (i.e., remove the target completely). This PR actually goes in the opposite direction.  (cc: @MikeSpreitzer ) .
2. It seems strange to me that 2 lines of yq replaced the whole script. Doesn't it mean that the script itself could have been just a couple of lines as well ?

**Comment by francostellari** [comment]
> @francostellari the previous script also had cleanup-tags. Isn't it being used anymore ?

I could not determine any impact of the cleanup in the final yaml.
@pdettori can you help me understand what it needs to do?

**Comment by francostellari** [comment]
> @francostellari one high level comment.
> 
> 1. In general I think we should reduce the usage of Makefile phony targets with complex logic in them and move instead to use utility scripts (i.e., remove the target completely). This PR actually goes in the opposite direction.  (cc: @MikeSpreitzer ) .
> 2. It seems strange to me that 2 lines of yq replaced the whole script. Doesn't it mean that the script itself could have been just a couple of lines as well ?

On the other hand putting 2 lines into a script, seems a bit excessive.
Let me know what you agree on, no difference from my point of view.

**Comment by MikeSpreitzer** [comment]
Yes, the old script could have been just two invocations of yq. I get the impression that the script we see today is the result of some evolution in the author's thinking, and he has not had time to fully tidy it up; I think that the litany of complaints in #1756 is evidence of this. I think that the bash function named `cleanup_helm_tags` in the old script is a vestige of old thinking. I think @francostellari's replacement here is much simpler and has no need of that bash function.

Since I last ranted about phony bash targets I have realized one complication in getting rid of them. When a phony target depends on a non-phony target, replacing that phony target with a script means that either the script invokes `make` on the non-phony dependency or invokes a script that has make-like functionality. Neither is entirely pleasing. In any case, this PR is not introducing any phony `make` targets, so I do not think we need to hold up this PR for that general concern.

**Review by MikeSpreitzer — COMMENTED** [review]
/lgtm
/approve
Thanks!

File: Makefile [hunk 1]
```diff
@@ -182,9 +182,10 @@ kind-load-image:
 .PHONY: chart
 chart: manifests kustomize
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(shell echo ${IMG} | sed 's/\(:.*\)v/\1/')
-	$(KUSTOMIZE) build config/default > chart/templates/controller-manager.yaml
-	scripts/add-helm-code.sh add
-
+	$(KUSTOMIZE) build config/default \
+		| yq '. | select(.kind == "ClusterRole*").metadata.name |= "{{.Values.ControlPlaneName}}-" + .' \
```
Comment (MikeSpreitzer) [inline Makefile:186]: Is `. |` a no-op in `yq`?

Comment (MikeSpreitzer) [reply Makefile:186]: Yes, it is a no-op. This is not a blocker.
/lgtm

**Pull request merged**